api/v1/configurationpolicy_types_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package v1
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestRecordDiffWithDefault(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		testName   string
+		apiVersion string
+		kind       string
+		recordDiff RecordDiff
+		expected   RecordDiff
+	}{
+		{
+			testName:   "Namespace-recordDiff-unset",
+			apiVersion: "v1",
+			kind:       "Namespace",
+			expected:   RecordDiffInStatus,
+		},
+		{
+			testName:   "Namespace-recordDiff-set",
+			apiVersion: "v1",
+			kind:       "Namespace",
+			recordDiff: RecordDiffLog,
+			expected:   RecordDiffLog,
+		},
+		{
+			testName:   "Configmap-recordDiff-unset",
+			apiVersion: "v1",
+			kind:       "ConfigMap",
+			expected:   RecordDiffSensored,
+		},
+		{
+			testName:   "Secret-recordDiff-unset",
+			apiVersion: "v1",
+			kind:       "Secret",
+			expected:   RecordDiffSensored,
+		},
+		{
+			testName:   "Secret-recordDiff-set",
+			apiVersion: "v1",
+			kind:       "Secret",
+			recordDiff: RecordDiffInStatus,
+			expected:   RecordDiffInStatus,
+		},
+		{
+			testName:   "OAuthAccessToken-recordDiff-unset",
+			apiVersion: "oauth.openshift.io/v1",
+			kind:       "OAuthAccessToken",
+			expected:   RecordDiffSensored,
+		},
+		{
+			testName:   "OAuthAuthorizeTokens-recordDiff-unset",
+			apiVersion: "oauth.openshift.io/v1",
+			kind:       "OAuthAuthorizeTokens",
+			expected:   RecordDiffSensored,
+		},
+		{
+			testName:   "Route-recordDiff-unset",
+			apiVersion: "route.openshift.io/v1",
+			kind:       "Route",
+			expected:   RecordDiffSensored,
+		},
```

Comment:
```suggestion
		},
		{
			testName:   "Route-recordDiff-unset",
			apiVersion: "route.openshift.io/v1",
			kind:       "Route",
			recordDiff: RecordDiffLog,
			expected:   RecordDiffSensored,
		},
```

Add this test too?

Reply:
If the user explicitly sets `recordDiff` we want to honor that. `RecordDiffSensored` is just to detect when sensitive data is present and we don't want to store the diff in the status by default.